### PR TITLE
New command structure, automatic command loader and help command

### PR DESCRIPTION
**Pull request by zaosoula**
New command structure with usage, description, aliases, and more.
The old command structure is still working.

The help command is working and gets generated automatically.

### CODE REVIEW

File: commands/echo/index.js [hunk 1]
```diff
@@ -1,3 +1,8 @@
...
-  opts.event.channel.send(opts.argsRaw.join(' '));
-}
+module.exports = {
+  description: 'Reply with the arguments',
+  usage: 'string...',
+  aliases: ['print', 'say'],
+	execute(opts) {
+    opts.event.channel.send(opts.argsRaw.join(' '));
+	},
+};
```
Comment (zaosoula) [inline commands/echo/index.js:8]: On this file, you can see the change between the v1 and v2 of the command structure

File: index.js [hunk 2]
```diff
@@ -23,38 +25,45 @@ client.once('ready', () => {
...
+  } else {
+    let opts = {
+      event,
+      args,
+      argsRaw,
+      client,
+      Discord,
+      commands,
+      Config
+    }
```
Comment (zaosoula) [inline index.js:63]: @Fabio-web  *opts* argument now pass Discord.js, commands, and Config objects to your command so no need to require Discord.js inside your file anymore.

File: utils/load-commands/index.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    commandObject = Object.assign({}, defaultCommandObject);
+    commandObject.name = name;
+
+    if (typeof required === 'function') {
+      Object.assign(commandObject, {
+        execute: required
+      });
+    } else if (typeof required === 'object') {
+      Object.assign(commandObject, required);
+    }
```
Comment (zaosoula) [inline utils/load-commands/index.js:40]: This is the part of the loader which manage the two different command structure

**Review by Maxi-Mega — APPROVED** [review]

File: index.js (hunk 2 above)
Comment (Fabio-web) [reply index.js:63]: Thank you 👍

**Review by Fabio-web — COMMENTED** [review]
👍

**Review by Fabio-web — APPROVED** [review]

**Pull request merged**